### ISSUE TITLE
Facilitate implementation of <Alt>, <Meta> etc. keyboard triggers

### ISSUE THREAD

**Issue by smeech** [federico-terzi/espanso#1933]
We have many requests to extend Espanso triggers to keyboard combinations including \<Alt> and \<Meta> etc. keys.

Operating systems have their own keybinding facilities which could be linked to commands such as `espanso match exec -t <trigger>`, but Espanso attempts to delete the trigger phrase, which doesn't exist when called in this manner, and therefore deletes the preceding text instead.

Currently, the best workaround I have found is to bind a key combination to a command using an external utility, e.g. `xdotool sleep 0.2 type <trigger>`, which invokes Espanso indirectly.

An additional parameter to `espanso match exec -t` which suppresses the trigger deletion, would enable anyone to link such a command to a keybinding and invoke it directly with that key-combination.

#1259, #1188, #1932,

**Comment by DoctorSubtilis** [comment]
Interesting. I have Linux and xdotool, but I don't understand how to do. Maybe you are speaking of a modification of espanso code? And not about something we can do?

**Comment by smeech** [comment]
Yes, my idea requires modification of the codebase, but I'm hoping it's simple enough to implement.

In the meantime, you can use the `xdotool` command I mentioned above. Happy to help if you need it.

**Comment by percit** [comment]
Yeah, would be lovely to have this instead of having one tool for alt shortcuts and another for other text application

**Comment by smeech** [comment]
Also #1555 and https://github.com/espanso/espanso/issues/1996#issuecomment-2222597637

**Comment by phly95** [comment]
The problem with xdotool is that it's platform specific. If you are sharing a script across a team with mixed OSs, then it would be nice to have a cross platform way of simulating keypresses.

For example, say I want to execute javascript when a phrase is typed in. On Windows, what I do is set a javascript bookmarklet to use a keyword (firefox feature), and then I can simulate the following keypresses in autohotkey:

1. Press Ctrl+L
2. Type the keyword into the address bar
3. Press enter.

Then the browser can execute javascript, allowing the script to grab info directly from the webpage, or trigger actions on the webpage, such as sending an email when the web app does not already have a keyboard shortcut, or selecting an email template automatically with a hotkey.

A guide for setting this type of keypress simulation workflow up would be helpful.

**Comment by smeech** [comment]
Yes, `xdotool` is a hack, as is [this](https://pastebin.com/6VfaxnMG) Python script, which *might* work in Windows.

That's precisely why I would like a change in the code to enable `espanso match exec -t <trigger>` *without* Espanso attempting to delete the (non-existent) trigger phrase.

The scenario you describe, however, can be done with Espanso if it's just a case of replacing the text in a browser address-bar, although I suspect you're describing something more sophisticated!

**Comment by phly95** [comment]
I'm describing being able to automatically navigate to the address bar when it is not selected, then type in text into the address bar.

**Comment by smeech** [comment]
In that instance, there is a <ctrl+l> code (0x0C) which you can use, e.g.:
```yml
  - trigger: "\x05" # <ctrl-e>
    replace: "\x0Csomething\n"
```
but Espanso is not going to be as capable as AutoHotKey (which I used for many years) and the `\n` doesn't seem to work in this example.

**Comment by phly95** [comment]
Neat! I think that might have been what I was looking for. I'll give it a try once I have the chance. Autohotkey was one of the last things I really needed from Windows.

For a python script of mine, I do want to get away from pyautogui and use something that works on Wayland, so maybe the solution involves some integration with espanso.

**Comment by smeech** [comment]
#1996 relates.
